### ISSUE TITLE
Vasprun().final_energy does not report energy(sigma->0) for VASP 6 & Vasprun and Outcar final energies are inconsistent

### ISSUE THREAD

**Issue by arosen93** [materialsproject/pymatgen#2367]
@shyuep, @mkhorton: There is a bug with the `final_energy` property in `Vasprun` for VASP 6+, and the `final_energy` property in `Outcar` is somewhat unintuitive.

## Background
For the attached VASP 6 files below, there are the three types of energies in the output for the final step.

Vasprun.xml:
```
   <i name="e_fr_energy">    -33.16377435 </i>
   <i name="e_wo_entrp">    -33.15237262 </i>
   <i name="e_0_energy">    -33.15807349 </i>
```

OUTCAR:
```
  free  energy   TOTEN  =       -33.16377435 eV

  energy  without entropy=      -33.15237262  energy(sigma->0) =      -33.15807349
```

Here is a quote from the VASP manual in the [ISMEAR section](https://www.vasp.at/wiki/index.php/ISMEAR) to motivate what I'm going to discuss below, although getting the sigma->0 extrapolated energy is important regardless of ISMEAR.

> The Gaussian smearing method leads to very reasonable results in most cases. Within this method **it is necessary to extrapolate from finite SIGMA results to SIGMA=0 results**. You can find an extra line in the OUTCAR file: energy( SIGMA→0 ), giving the extrapolated results.

## Vasprun
For the attached files, the `Vasprun` prints out the `e_wo_entrp` as the `final_energy`. It's because of the codeblock below, where it detects that `e_wo_entrp` and `e_0_energy` are different and takes the former.

It's not particularly clear to me, however, why it's taking the former when you'd almost universally want the energy in the sigma --> 0 limit (there are no vdW corrections here despite the warning message). **NOTE**: Turns out the code is not what it looks like at first glance. It's to fix a confusing `vasprun.xml` bug in VASP 5 (see comment below), but it ends up breaking in VASP 6.

https://github.com/materialsproject/pymatgen/blob/ae3f1b77d75b397735ee7cf16fcd77d8f487d85e/pymatgen/io/vasp/outputs.py#L621-L629

## Outcar
To add to the confusion a bit, the `Outcar` prints out the `e_fr_energy` instead as the `final_energy`. Again, I would have naively assumed that it would output the `e_0_energy`. There are few situations where I'd actually want `e_fr_energy`. Either way, it is not great that it is a different `final_energy` than that generated with `Vasprun`.

This can be traced back to the following line, which looks for `TOTEN` and then uses that energy for `final_energy`.
https://github.com/materialsproject/pymatgen/blob/ae3f1b77d75b397735ee7cf16fcd77d8f487d85e/pymatgen/io/vasp/outputs.py#L1795

## Example
```python
from pymatgen.io.vasp import Vasprun,Outcar

vr = Vasprun('Vasprun.xml.gz')
print(vr.final_energy) # -33.15237262
out = Outcar('OUTCAR.gz')
print(out.final_energy) # -33.16377435

# Note: The energy(sigma->0) is -33.15807349.
```
[OUTCAR.gz](https://github.com/materialsproject/pymatgen/files/7919351/OUTCAR.gz)
[vasprun.xml.gz](https://github.com/materialsproject/pymatgen/files/7919352/vasprun.xml.gz)

**Comment by shyuep** [comment]
I think the reason is mostly historical. And I am fine with changing everything to default to e_0_energy unless @mkhorton has any concerns.

**Comment by arosen93** [comment]
Good to know.

I did some digging into this, and I now see what was trying to be done. For `Vasprun`, it appears this was a workaround to address a bug in VASP 5.x.x, which had the correct `e_0_energy` in the OUTCAR but _not_ in certain parts of the `vasprun.xml` when including vdW corrections. This is resolved in VASP 6+. Unfortunately, the `Vasprun` code right now, as a side-effect, also selects the "incorrect" energy value for calculations with ISMEAR = 0 (even without vdW) because it detects a difference in `e_0_energy` and `e_wo_entrp`, thinks it might be the vdW issue, and so picks `e_wo_entrp` out of caution. 

The `Outcar` scenario seems to just be purely a historical decision.

I have created a test suite of several different runs with VASP 5.x and VASP 6.x with and without vdW. I will use this to propose a fix to `Vasprun` that addresses all this. More details to come.
